src/Domain/Block.cpp

```diff
@@ -23,7 +23,7 @@ Block<VolumeDim>::Block(
     std::unique_ptr<domain::CoordinateMapBase<
         Frame::BlockLogical, Frame::Inertial, VolumeDim>>&& stationary_map,
     const size_t id,
-    DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors,
+    DirectionMap<VolumeDim, BlockNeighbors<VolumeDim>> neighbors,
```

Comment:
When you rebase you can assert here that the number of neighbors in each dimension is consistent with the topology of the block

Reply:
this is done in the next PR which also correctly sets the number of external boundaries based on the topology and input neighbors